### ISSUE TITLE
Missing "read" dependency

### ISSUE THREAD

**Issue by dmcassel** [marklogic/node-client-api#205]
Examples/setup.js requires the npm "read" package, but this isn't included in package.json. 

```
$ node examples/setup.js 
module.js:338
    throw err;
          ^
Error: Cannot find module 'read'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/dcassel/tmp/myproject/node_modules/marklogic/etc/test-setup-prompt.js:16:14)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/dcassel/tmp/myproject/node_modules/marklogic/examples/setup.js:21:22)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
```

**Comment by dmcassel** [comment]
I see the package is included in the devDependencies list, so I guess the expectation is that someone using the node-client-api as part of some other project won't be running the examples. Someone tried running them while going through a tutorial and setup.js broke. Worth moving from devDependencies to dependencies?

**Comment by ehennum** [comment]
I agree that the examples should be trivial to run. I'm not sure applications should have to install the Node.js API examples as their runtime dependency. Let's talk about alternatives.

**Comment by dmcassel** [comment]
If the examples weren't included when someone "npm installs" marklogic, then yeah, the ability to run them doesn't matter. If they're going to be included (as they are now), they should be runnable. Solution could go either direction.

**Comment by Radu3000** [comment]
I am also interested to see it working.  just trying marklogic for now and will need nodejs integration soon.  

P.S. Is it possible to change the examples use cases from ".bind('sex'))" to a different domain use case altogether - i.e.without an innuendos? 

Thank you!

**Comment by ehennum** [comment]
From a quick look, the only obvious solution was spinning off the examples as a separate project.

**Comment by jmakeig** [comment]
@wooldridge, is this still an issue?

**Comment by ehennum** [comment]
On balance, the best solution is to add a .npmignore file that keeps the examples directory in the GitHub repository but drops it from the npm installable distribution:

https://npm.github.io/publishing-pkgs-docs/publishing/the-npmignore-file.html

**Comment by ehennum** [comment]
If it's too late for the client release supporting the 10.0-8 server release, this fix could be delayed to the 10.0-9 timeframe.
